### PR TITLE
Add Beachball

### PR DESCRIPTION
**Pull request by aeunms**
[Beachball](https://microsoft.github.io/beachball/)!  
Beachball is a semantic version bumper that also has an automated tool to ask devs to log changes in a simple CLI manner.

How-to is documented in CONTRIBUTING.md. v2 folks can find the doc and discussion for this in our Engineering channel in Teams.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (erinha) [inline CONTRIBUTING.md:1]: @TrevorJoelHarris  I know you were looking into making changes to Contributing guidelines. Tagging in case you have feedback based on that parallel effort.

File: CONTRIBUTING.md
Comment (erinha) [inline CONTRIBUTING.md:41]: Nit: can you use "contributors" instead of "devs" please?

File: CONTRIBUTING.md
Comment (erinha) [inline CONTRIBUTING.md:43]: I think it's clearer to say: "The TeamsJS Client SDK v2 `2.0-preview` branch contains..." since we're in the same repo.

File: CONTRIBUTING.md
Comment (erinha) [inline CONTRIBUTING.md:47]: nit: Can you capitalize the "w" in "We won't be allowing..."?

Comment (erinha) [reply CONTRIBUTING.md:47]: Actually, can you rephrase it as "The Prerelease option is not allowed in this repo at this time."

File: packages/teams-js/README.md
Comment (erinha) [inline packages/teams-js/README.md:3]: I think it could be clearer to link directly to the CHANGELOG file, e.g.:

"For breaking changes, please refer to our [CHANGELOG](./CHANGELOG.md) in the current `<root>/packages/teams-js` directory of the `2.0-preview` branch."

(Note: I didn't validate that link works in my example)

File: enforceBeachball.js
Comment (erinha) [inline enforceBeachball.js:9]: The script command "yarn generate-change-files" actually calls "yarn beachball change". Is "yarn beachball change" the same as "yarn beachball"? I feel like perhaps all our instructions should just include one command to run to keep it simple (e.g., "yarn generate-change-files").

File: packages/teams-js/CHANGELOG.md
Comment (erinha) [inline packages/teams-js/CHANGELOG.md:1]: I think we had talked about changing the format of the CHANGELOG so that it would match how beachball adds comments to it. I don't think this has be done as part of this change. Are you planning to do that still?

**Review by erinha — CHANGES_REQUESTED** [review]
Super excited for this change, thanks! Left a few questions/suggestions.

File: CONTRIBUTING.md
Comment (aeunms) [reply CONTRIBUTING.md:41]: Done, will be included in the next commit(s)!

File: CONTRIBUTING.md
Comment (aeunms) [reply CONTRIBUTING.md:43]: Done, will be included in the next commit(s)! (I ended up tweaking it so there is no v2 in the sentence)

File: CONTRIBUTING.md
Comment (aeunms) [reply CONTRIBUTING.md:47]: Done, will be included in the next commit(s)!

File: packages/teams-js/README.md
Comment (aeunms) [reply packages/teams-js/README.md:3]: Great idea, I've linked it in my local commit and will push soon :)

File: enforceBeachball.js
Comment (aeunms) [reply enforceBeachball.js:9]: Good catch! I initially started out with `yarn beachball change` but slowly changed to `yarn beachball` because it's the same thing but shorter. I thought it would make more sense to use `yarn beachball` but in case people forgot the command I included an alias for it in the scripts- but now I'm thinking maybe that's not necessary. I'm going to replace the instances with `yarn changefile`. I think it's super clear what it's doing with that name and it's not much longer than `yarn beachball`.

File: packages/teams-js/CHANGELOG.md
Comment (aeunms) [reply packages/teams-js/CHANGELOG.md:1]: I did go through and add a # to all headers as part of this change, it's just hard to tell because it's also been moved so it just shows as a whole new file 😋

File: enforceBeachball.js
Comment (ramourt) [inline enforceBeachball.js:3]: What is the killProcessCommand.txt file? Can this run without it?

File: enforceBeachball.js
Comment (ramourt) [reply enforceBeachball.js]: Yeah `changefile` is ok. `beachball:change` or `change` are the most common commands I've seen in repos

File: packages/teams-js/CHANGELOG.md
Comment (ramourt) [reply packages/teams-js/CHANGELOG.md:1]: I'm planning to hook up all of this stuff to the github release workflows we have once we merge this. 
Planning to change the 2.0.0-beta.3 ones to come in through changefiles. The rest I can just change manually if we want the older versions to have the same bulleted list formatting.

File: CONTRIBUTING.md
Comment (ramourt) [reply CONTRIBUTING.md]: Can we remove mentioning the prerelease option altogether? We won't be using it now or any time later.

File: enforceBeachball.js
Comment (ramourt) [inline enforceBeachball.js]: Why does this need the killProcessCommand.txt file? What's in the file?

**Review by ramourt — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md
Comment (ramourt) [reply CONTRIBUTING.md]: And it won't really show up for people since we have it disabled in the beachball config

File: enforceBeachball.js
Comment (aeunms) [reply enforceBeachball.js:3]: It just contains ^C. When you run yarn beachball, it gives one of two responses: it either says you don't need a changefile and terminates or it prompts you for an answer. The ^C in the killProcessCommand.txt file makes it so it doesn't hang there forever waiting for an answer and time out the check. I experimented with timeouts and having a timeout of 3~5 seconds also works, but I thought this made more sense. What do you think is better? (Or maybe you have a better option?)

File: CONTRIBUTING.md
Comment (aeunms) [reply CONTRIBUTING.md]: Yes for sure. Upcoming in next commit.

File: enforceBeachball.js
Comment (aeunms) [reply enforceBeachball.js]: I think this is a duplicate comment- will condense into the other thread.

File: CONTRIBUTING.md
Comment (aeunms) [reply CONTRIBUTING.md]: Updated

File: CONTRIBUTING.md
Comment (aeunms) [reply CONTRIBUTING.md]: Updated

File: packages/teams-js/README.md
Comment (aeunms) [reply packages/teams-js/README.md:3]: Updated

File: enforceBeachball.js
Comment (aeunms) [reply enforceBeachball.js]: Updated

File: enforceBeachball.js
Comment (aeunms) [reply enforceBeachball.js]: We discussed offline and thought the timeout was better :) switching to timeout

**Review by ramourt — APPROVED** [review]
:shipit:

File: packages/teams-js/CHANGELOG.md
Comment (aeunms) [reply packages/teams-js/CHANGELOG.md:1]: Speaking of hooking up github release workflows- I found that Beachball doesn't execute the postbump hook if the change type was "None". So we'll need to include deleting the JSON changelog somewhere in release or a pipeline but I'm not sure which ticket to file that in, does anyone know?

Comment (aeunms) [reply packages/teams-js/CHANGELOG.md:1]: Also, I don't know if that's intended behavior or a bug. I haven't found any official docs on it. I just found out from testing it

File: packages/teams-js/README.md
Comment (erinha) [reply packages/teams-js/README.md:3]: I don't actually see the link in the code here. It's okay if you decided not to add the link, but if you did, it's not showing for me.

File: CONTRIBUTING.md
Comment (TrevorJoelHarris) [reply CONTRIBUTING.md:1]: Yes, I did add some internal only guidelines for contributing to our wiki. I think for now there's nothing to do here. Ultimately I'd like to take a swing at standardizing the internal and external guidance but I think that will involve some careful work that requires us making sure we've crossed all our t's and dotted all our i's.

This seems very targetted to "how to check in to this repo" which seems okay. I don't think there's any need to converge the two before angela checks in

File: CONTRIBUTING.md
Comment (erinha) [inline CONTRIBUTING.md:59]: Maybe "Our pipelines will check to see if you generated a changefile and will fail if you forgot. If they do, please create the changefile as per the steps listed and update the content accordingly."

Comment (aeunms) [reply CONTRIBUTING.md]: Done!

**Review by ramourt — DISMISSED** [review]

File: packages/teams-js/README.md
Comment (aeunms) [reply packages/teams-js/README.md:3]: Very weird. It's also not showing for me in the GitHub UI but it's there in my VS Code. 
image
Could you do me a favor and see what it looks like if you pull my branch?

Comment (aeunms) [reply packages/teams-js/README.md:3]: Wait.... I have the wrong README. lol I'm going to fix it.

**Review by erinha — APPROVED** [review]
LGTM!

**Pull request merged**
